src/integration/resources/resources.go

```diff
@@ -49,12 +50,15 @@ func SetupCluster(cluster M3Resources, opts *ClusterOptions) error { // nolint:
 	logger := iOpts.Logger().With(zap.String("source", "harness"))
 	hosts := make([]*admin.Host, 0, len(cluster.Nodes()))
 	ids := make([]string, 0, len(cluster.Nodes()))
-	for _, n := range cluster.Nodes() {
+	for i, n := range cluster.Nodes() {
 		h, err := n.HostDetails(9000)
 		if err != nil {
 			logger.Error("could not get host details", zap.Error(err))
 			return err
 		}
+		if opts.NumIsolationGroups > 0 {
+			h.IsolationGroup = fmt.Sprintf("isogroup-%d", int32(i)%opts.NumIsolationGroups)
```

Comment:
I see the IsolationGroup is added here. Maybe add a comment in the HostDetails() function about the missing isolation group info?

Reply:
I'm gonna refactor the HostDetails method in a future PR so I'm OK with just leaving it as-is for now.